### PR TITLE
feat: image-and-card-animation

### PR DESCRIPTION
**Pull request by AnirudhMemani**
## Summary

+ Added scale (zoom-in) animation to the blog card images on hover.

## Description

+ Implemented a smooth scale animation effect to enhance user experience when hovering over the blog card images.
+ This animation adds visual appeal and interactivity to the blog section, drawing attention to the content and encouraging user engagement.
+ Also, added onClick scale animation to the card.

https://github.com/krishnaacharyaa/wanderlust/assets/46074384/d5fae1bf-a2ce-41cf-87cb-2efd9b5254d2

## Comments

+ Added subtle image animation on hover
+ Added cursor-text on headings and sub-headings

### CODE REVIEW

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Please check this out and let me know what you think.

File: frontend/src/components/featured-post-card.tsx [hunk 1]
```diff
@@ -13,15 +13,15 @@ export default function FeaturedPostCard({
...
       data-testid={testId}
     >
-      <div className="w-full sm:w-1/3">
+      <div className="w-full overflow-hidden sm:w-1/3">
         <img
           src={post.imageLink}
           alt={post.title}
-          className="h-48 w-full rounded-lg object-cover shadow-lg sm:h-full"
+          className="h-48 w-full rounded-lg object-cover shadow-lg transition-transform duration-300 ease-in-out sm:h-full md:group-hover:scale-125"
         />
```
Comment (krishnaacharyaa) [inline frontend/src/components/featured-post-card.tsx:25]: Can we reduce a scale little bit, to make it look subtle

File: frontend/src/components/post-card.tsx [hunk 2]
```diff
@@ -9,16 +9,21 @@ export default function PostCard({ post, testId = 'postcard' }: { post: Post } &
...
-          src={post.imageLink}
-          alt={post.title}
-          className="h-48 w-full rounded-t-lg object-cover"
-        />
+        <div className="h-48 w-full overflow-hidden">
+          <img
+            src={post.imageLink}
+            alt={post.title}
+            className="h-full w-full rounded-t-lg object-cover transition-transform duration-300 ease-in-out md:group-hover:scale-125"
+          />
```
Comment (krishnaacharyaa) [inline frontend/src/components/post-card.tsx:25]: Same all the places, let it be very light and subtle....

**Review by krishnaacharyaa — COMMENTED** [review]
Thank you for the PR, it is interesting addition :) Can we have it a bit subtle, And please update the video after done, because of the deployment issues, Can't preview the changes live, If possible can you create a new PR with the latest code? That would help and avoid some confusion because of the recent code merge. 

p.s: Remember in the backend env you would have to add CORS_ORIGIN, you can follow the env.sample

**Comment by krishnaacharyaa** [comment]
@AnirudhMemani can you kindly revise the PR with the comments

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Sorry, I was busy with some work. I will do it now!

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Check it out, please excuse me if I have not followed the right way to raise PR since this is my very first PR ever !!

**Comment by krishnaacharyaa** [comment]
No problem @AnirudhMemani, I guess 120-150 range is good, this subtle is not working is what I am feeling, and delay we have to reduce, there is little delay we have set for the zoom in transition to happen lets keep it less, lets make it dramatic.

About commit, it is totally fine, just make sure you write proper description, 
`feat: image-animation` could have been `fix: Fixed the scaling of the image by reducing as requested.`
Something which actually conveys the problem

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Actually, It was at 125 the first time. Should I put it back to 125 but remove the transition duration altogether?

**Comment by krishnaacharyaa** [comment]
It will be mostly trial and error, can't exactly say.
Chota call can we have? For 5mins....
Else try keeping 140 and remove transition delay. Let's see

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Can we get on a zoom/google meet call because I need to record, remove audio and compress the video every time I have to share it here?

**Comment by krishnaacharyaa** [comment]
Sure...
Discord?

**Comment by krishnaacharyaa** [comment]
@AnirudhMemani , can you please join the discord from the readme file and then we'll finish this PR and get it merged asap over a call

**Comment by AnirudhMemani** [comment]
> @AnirudhMemani , can you please join the discord from the readme file and then we'll finish this PR and get it merged asap over a call

Joining discord in 5

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Done, check it out

**Comment by krishnaacharyaa** [comment]
Looks good, but lets not do in the scaling-animation.ts way, seems like a overhead/complicated thing, lets add a normal custom class in the @layer
https://tailwindcss.com/docs/adding-custom-styles#using-css-and-layer

Something like
.scale-hover { ... }
.scale-click { ... }

And use sm: instead of md:

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Done

File: frontend/src/index.css [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+  .scale-hover:hover {
+    transform: scale(1.12);
+  }
+
+  @media (min-width: 640px) {
+    .sm\:scale-hover:hover {
+      transform: scale(1.12);
+    }
+  }
+
```
Comment (krishnaacharyaa) [inline frontend/src/index.css:15]: Why are we doing explicit declaration ? , `sm:scale-hover` should ideally work without the above @media code in my opinion, And if not, is it the right way? it is almost like your classname is `sm:scale-hover` you have removed significance of `sm` in your class `sm:scale-hover:hover`.

**Comment by krishnaacharyaa** [comment]
@AnirudhMemani kindly let me know once it is resolved lets get it merged

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Done. Something so simple has taken us so much time. I'm embarrassed.

**Comment by krishnaacharyaa** [comment]
> @krishnaacharyaa Done. Something so simple has taken us so much time. I'm embarrassed.

No worries :), it happens

File: frontend/src/layouts/header-layout.tsx
[the remark this answers is not in the document]
Comment (Aeromania) [reply frontend/src/layouts/header-layout.tsx:1]: @krishnaacharyaa Added

**Review by krishnaacharyaa — APPROVED** [review]
LGTM, Thank you for adding animations out of interest @AnirudhMemani :)
Much appreciated, Feel free to come up with further improvements if any. Happy to consider :)

**Comment by AnirudhMemani** [comment]
@krishnaacharyaa Thank you. Will do!

**Pull request merged**